### PR TITLE
eapply_user: combine sort for all dirs (bug 608880)

### PR DESCRIPTION
**Pull request by zmedico**
Combine the patch basenames from all matched directories into a
list, and apply them in POSIX sorted order.  This allows patches in
more-specific directories to override patches of the same basename found
in less-specific directories. An empty patch (or /dev/null symlink)
negates a patch with the same basename found in a less-specific
directory.

This behavior is much more flexible and intuitive than the previous one,
while remaining backward-compatible to some extent.

NOTE: The implementation uses an associative array, which requires bash
version 4 or later.

X-Gentoo-bug: 608880
X-Gentoo-bug-url: https://bugs.gentoo.org/608880

### CODE REVIEW

File: bin/phase-helpers.sh [hunk 1]
```diff
@@ -1094,23 +1094,44 @@ if ___eapi_has_eapply_user; then
...
 		done
 
+		if [[ ${#_eapply_user_patches[@]} -gt 0 ]]; then
+			while read -r -d '' f; do
+				f=${_eapply_user_patches[${f}]}
+				if [[ -s ${f} ]]; then
+					eapply "${f}"
+					applied=1
+				fi
+			done < <(printf -- "%s\0" "${!_eapply_user_patches[@]}" |
```
Comment (mrueg) [inline bin/phase-helpers.sh]: shellcheck:  ^-- SC1117: Backslash is literal in "\0". Prefer explicit escaping: "\\0".
https://github.com/koalaman/shellcheck/wiki/SC1117

Comment (zmedico) [inline bin/phase-helpers.sh]: Thanks, fixed now.

Comment (mgorny) [inline bin/phase-helpers.sh]: I'd dare say a more readable fix would be to use single quotes, i.e. `'%s\0'`.

Comment (zmedico) [inline bin/phase-helpers.sh]: Yeah that's better. Done now.

**Pull request merged**